src/Exceptions.jl

```diff
@@ -33,29 +33,33 @@ function try_with_timeout(f, shouldtimeout, delay, iftimeout=() -> nothing)
         notify(cond, f())
     catch e
         @debugv 1 "error executing f in try_with_timeout"
-        notify(cond, e)
+        isopen(timer) && notify(cond, e, error = true)
```

Comment:
Why do we need to check `isopen(timer)` here? It seems like it wouldn't matter whether it's open or not?

Reply:
`iftimeout()` may cause `f` to throw. In that scenario I'd rather pick up the `TimeoutError` in `wait(cond)` and swallow the exception from `f`